protocols/gossipsub/src/behaviour.rs

```diff
@@ -605,9 +606,10 @@ where
     pub fn publish(
         &mut self,
         topic: impl Into<TopicHash>,
-        data: impl Into<Vec<u8>>,
+        data: impl Into<Bytes>,
     ) -> Result<MessageId, PublishError> {
-        let data = data.into();
+        let data: Bytes = data.into();
```

Comment:
```suggestion
        let data = data.into();
```

Shouldn't need the type annotation if we take an `impl Into`.